src/Accelerators/NNPA/Conversion/ZHighToZLow/ZHighToZLow.cpp

```diff
@@ -205,15 +207,18 @@ Value insertAllocOrEmitZeroConstant(ArrayRef<IndexExpr> dims,
             /*value=*/nullptr,
             /*alignment=*/rewriter.getI64IntegerAttr(4096));
 
-    // Use an opaque attribute to store stickified data.
+    // Use an dense resource attribute to store stickified data.
     // Attribute type: tensor<sizeInBytes x i8>
-    int64_t sizeInBytes = getMemRefSizeInBytes(resType).getValue();
+    int64_t sizeInBytes = getMemRefSizeInBytes(resType).value();
     char *rawData = (char *)malloc(sizeInBytes);
     memset(rawData, 0, sizeInBytes);
-    OpaqueElementsAttr valueAttr =
-        OpaqueElementsAttr::get(stickifiedConstant.getOperation()->getDialect(),
-            RankedTensorType::get({sizeInBytes}, rewriter.getI8Type()),
-            StringRef(rawData, sizeInBytes));
+    DenseResourceElementsAttr valueAttr = DenseUI8ResourceElementsAttr::get(
+        RankedTensorType::get({sizeInBytes}, rewriter.getI8Type()),
+        stickifiedConstant.getOperation()
+            ->getDialect()
+            ->getNamespace(), // use the dialect as the blob "hint"
+        UnmanagedAsmResourceBlob::allocate(
+            ArrayRef(rawData, sizeInBytes), alignof(char)));
     stickifiedConstant.valueAttr(valueAttr);
     free(rawData);
```

Comment:
@sstamenova removing this `free` will lead to the correct result on NNPA. It is a simple fix, but I am not sure whether it causes memory leak or not.

Reply:
@tungld Thanks! I switched the blob from Unmanaged to Heap, so that it can allocate and free memory itself - this way we can still call `malloc/free` here, but the blob will make its own copy of the data and manage its lifetime. I think the new change should work.